### DISCUSSION TITLE
How can I use values in a jsonb field to determine what to do in a trigger function?

### DISCUSSION THREAD

**Discussion by peppermintbunny** [supabase/supabase#12450] [Questions]
My app is really in two parts but the two parts share the same authentication table viz auth.users. I want to set up a trigger and function to be triggered whenever a new user is added to the auth system.  John Meyers has a video showing how to set up a trigger+function to add a new record to public.profiles whenever a new auth.user is created.  I want to add a new record to public.patient_profile when raw_user_meta_data->>'kind' equals 'patient' and a new record to public.therapist_profile when raw_user_meta_data->>'kind' equals 'therapist'.  I need help is writing the if statement in the following function (maybe I need more help 😂):

begin
............if.raw_user_meta_data->>'kind' equals 'patient'..............
  insert into public.patient_profile(id,Email,raw_user_meta_data)
  values(new.id,new.Email,new.raw_user_meta_data)
..............elseif raw_user_meta_data->>'kind' equals 'therapist'.............
 insert into public.therapist_profile(id,Email,raw_user_meta_data)
  values(new.id,new.Email,new.raw_user_meta_data)
  return new;
end;

I suppose I should add that raw_user_meta_data column contains a jsonb object like: {"kind":"therapist","displayname":"James"}

Many thanks

**Comment by GaryAustin1** [comment]
For if then format:  https://www.postgresqltutorial.com/postgresql-plpgsql/plpgsql-if-else-statements/ 
 
 For your JSONB 
` if new.raw_user_meta_data->>'kind' = 'patient' `

Warning:  You are using raw_user_meta_data.  This is not secure, and should not be used for critical data.  The user can change this at anytime if they have a signed in access_token, using updateUser() call.

**Reply by peppermintbunny** [reply]
Thanks for the help and much thanks for the warning.  Which table would updateUser() act on?  I'm supposing auth.user since a user would have no knowledge of the existence of public.profile or in my case public.therapist_profile and pulic.patient_profile.

**Reply by GaryAustin1** [reply]
Yes. So if you don’t trigger on updates then your profile table will not change if they use updateUser.
